lib/request-proxy/index.js

```diff
@@ -221,6 +235,38 @@ RequestProxy.prototype.handleRequest = function handleRequest(head, body, cb) {
         }
     }
 
+    var whoami = ringpop.whoami();
```

Comment:
Pulling this block to its own function will over complicate the flow since there is a pre-emptive escape in the block which is essential.

I will add a comment above this block (and for that matter to the block above as well) explaining why this is executed in place and always, eg. it gives stats on how many requests are executed on wrong hosts if when consistency is not enforced.